src/services/activation/utils/index.js

```diff
@@ -0,0 +1,169 @@
+	}
+	return filteredEntries;
+};
+
+const validEntry = async (entry) => {
+	if (!entry) throw new NotFound('ACTIVATION_LINK_INVALID');
+	if (
+		Date.parse(entry.updatedAt) + 1000 * Configuration.get('ACTIVATION_LINK_PERIOD_OF_VALIDITY_SECONDS')
+		< Date.now()
+	) {
+		await deleteEntry(this, entry._id);
+		throw new BadRequest('ACTIVATION_LINK_EXPIRED');
+	}
+	if (entry.state !== STATE.notStarted || entry.state !== STATE.error) {
+		throw new BadRequest('ACTIVATION_LINK_INVALID');
+	}
+};
+
+/**
+ * Will lookup entry in a activation by userId
+ * @param {*} ref 					this
+ * @param {ObjectId} requestingId	ObjectId of User requesting data
+ * @param {ObjectId} userId			ObjectId of User (userId)
+ * @param {String} keyword			Keyword
+ */
+const lookupByUserId = async (ref, requestingId, keyword = null) => {
+	if (!requestingId) throw new Forbidden('Not authorized');
+
+	const entry = await (ref.app || ref).service('activationModel').find({ query: { account: requestingId } });
+	return lookupEntry(requestingId, entry, keyword);
+};
+
+/**
+ * Will lookup entry in a activation by entryId
+ * @param {*} ref 					this
+ * @param {ObjectId} requestingId	ObjectId of User requesting data
+ * @param {String} activationCode	activationCode
+ * @param {String} keyword			Keyword
+ */
+const lookupByActivationCode = async (ref, requestingId, activationCode, keyword = null) => {
+	if (!requestingId) throw new Forbidden('Not authorized');
+	if (!activationCode) throw SyntaxError('entryId required!');
+
+	const entry = await (ref.app || ref).service('activationModel').find({ query: { activationCode } });
+	return lookupEntry(requestingId, entry, keyword);
+};
+
+/**
+ * Will send email with informatrion from mail {email, subject, content}
+ * also sets mailSend and updatedAt for entry by entryId
+ * @param {*} ref				this
+ * @param {Object} mail			{email, subject, content}
+ * @param {ObjectId} entryId	ObjectId of entry
+ */
+const sendMail = async (ref, mail, entry) => {
+	if (!mail || !mail.email || !mail.subject || !mail.content || !entry) throw SyntaxError('missing parameters');
+
+	try {
+		// await ref.app.service('/mails')
+		// 	.create({
+		// 		email: mail.email,
+		// 		subject: mail.subject,
+		// 		content: mail.content,
+		// 	});
+
+		await ref.app.service('activationModel').patch({ _id: entry._id }, {
+			$set: {
+				mailSend: true,
+			},
+		});
+	} catch (error) {
+		if (!entry.mailSend) deleteEntry(ref, entry._id);
+		throw new Error('Can not send mail with activation link');
+	}
+};
+
+const sanitizeEntries = (enties, validKeys) => {
+	(enties || []).forEach((entry) => {
+		let data = {};
+		if ('quarantinedObject' in entry) {
+			data = getQuarantinedObject(entry.keyword, entry.quarantinedObject);
+			delete entry.quarantinedObject;
+		}
+		Object.keys(entry).forEach((key) => validKeys.includes(key) || delete entry[key]);
+		entry.data = data;
+	});
+	return enties;
+};
+
+const getUser = async (ref, userId) => {
```

Comment:
```suggestion
const getUser = async (ref, userId, req=null) => {
```
if (req.params.user) return user